src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs

```diff
@@ -307,77 +374,25 @@ private void WriteSchemaObjectExampleProperties(CodeWriter writer, ObjectType so
                         var paramValue = FindPropertyValue(sot, ev, targetProperty!);
                         if (paramValue is not null)
                         {
-                            FormattableString newVariableName = $"{variableName}.{targetProperty.Declaration.Name}";
+                            FormattableString newVariableName = $"{PropertyVaraibleName(variableName, targetProperty)}";
                             if (targetProperty.Declaration.Name == "Tags" && targetProperty.ValueType.Name == "IDictionary")
                             {
-                                CreateTagWriterDelegate(newVariableName, targetProperty.ValueType, paramValue);
+                                AddTagWriterDelegate(newVariableName, targetProperty.ValueType, paramValue);
+                                consumedProperties.Add(targetProperty);
                             }
-                            else
+                            else if (!targetProperty.IsSinglePropertyObject(out var innerProperty))
                             {
                                 writer.Append($"{targetProperty.Declaration.Name:D} = ");
                                 WriteExampleValue(writer, targetProperty.ValueType, paramValue!, newVariableName);
                                 writer.Append($", ");
+                                consumedProperties.Add(targetProperty);
                             }
-                            consumedProperties.Add(targetProperty);
                         }
                     }
                 }
             }
         }
 
-        internal readonly struct JsonRawValue
```

Comment:
Instead of moving this to `CodeModelPartial`, I believe we could create a `Model` namespace inside the `MgmtTest` directory and put it there